### PR TITLE
edit functions.py

### LINKED ISSUE #12292: Fatal Python error: Cannot recover from stack overflow
Happens when executing this:
```
 mak  ~  python3
Python 3.5.2 (default, Sep 14 2016, 11:28:32) 
[GCC 6.2.1 20160901 (Red Hat 6.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import *
>>> var('x')
x
>>> f = diff(floor(x), x)
>>> f
Derivative(floor(x), x)
>>> N(f.subs(x, 3))
Fatal Python error: Cannot recover from stack overflow.

Current thread 0x00007f6fded09700 (most recent call first):
  File "/usr/lib/python3.5/site-packages/sympy/core/basic.py", line 299 in __eq__
  File "/usr/lib/python3.5/site-packages/sympy/core/cache.py", line 93 in wrapper
  File "/usr/lib/python3.5/site-packages/sympy/core/expr.py", line 141 in __mul__
  File "/usr/lib/python3.5/site-packages/sympy/core/decorators.py", line 118 in binary_op_wrapper
  File "/usr/lib/python3.5/site-packages/sympy/core/decorators.py", line 77 in __sympifyit_wrapper
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 520 in _eval_derivative
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1143 in __new__
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1273 in doit
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1571 in doit
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  File "/usr/lib/python3.5/site-packages/sympy/core/function.py", line 1574 in evalf
  ...
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by ghost**
add new function in subs class for solving derivative expression for floor and ceiling problem

<!-- If this pull request fixes an issue please indicate which issue by typing "Fixes #12292 " below.>

### CODE REVIEW

File: sympy/core/function.py [hunk 1]
```diff
@@ -1769,6 +1769,40 @@ def _eval_as_leading_term(self, x):
...
+        """this function is used to Solve the floor and ceiling derivatives
+         both = True if the expression contain both floor and ceiling
+         both = False if the expression contain only floor or ceiling expressions"""
+
+        from sympy.printing import StrPrinter
+        from sympy import floor
+        from sympy import ceiling
+        from sympy import sympify
+        total = 0
+        m = str(self).replace(" + ", "!")
```
Comment (smichr) [inline sympy/core/function.py]: I suspect this can be solved without strings. What (in words) are you trying to do?

**Comment by gxyd** [comment]
@clavoirr can you please add some doctests and test cases for this functionality? I don't clearly understand what you are trying to do here.

**Comment by gxyd** [comment]
You should also update your pull request title to a little more descriptive. See this https://github.com/sympy/sympy/wiki/Development-workflow#writing-pull-request-title-and-description

**Comment by gxyd** [comment]
This PR has been inactive for quite sometime, so we are going to close it and open an issue to reference it. We try to keep our pull request queue small and focused on active work.  We encourage you to reopen the pull request if and when you continue to work on this. Please contact us if you have any questions.

Thanks for contributing.

see https://github.com/sympy/sympy/wiki/Closing-pull-requests for our policies on closing pull request.

We already have issue https://github.com/sympy/sympy/issues/9230 for this.

**Pull request closed without merging**